packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -460,10 +472,14 @@ class _HeroFlightManifest {
   CurvedAnimation? _animation;
 
   Animation<double> get animation {
+    final Curve reverseCurve = (type == HeroFlightDirection.push) ?
```

Comment:
Can you convert this entire section into a if else? it will probably be more readable given all the property in the curved animation is inline if else now.